public/js/index.js

```diff
@@ -499,6 +505,71 @@ $(window).on('error', function () {
   // setNeedRefresh();
 })
 
+function checkParametr (isLogin, permission) {
+  if (typeof isLogin !== 'boolean' || !permission) {
+    throw new Error('one or more parametr is incorrect')
+  } else return allowVisibleSource(isLogin, permission)
+}
+
+function replaceUrl (url) {
+  const urlHasEditOrBoth = /\?edit|\?both/
+  if (urlHasEditOrBoth.test(url)) {
+    const newUrl = url.toString().replace(urlHasEditOrBoth, '?view')
+    window.location.replace(newUrl)
+  }
+}
+
+function allowVisibleSource (isLogin, permission) {
+  switch (permission) {
+    case 'freely':
+      blockSourceView = false
+      break
+    case 'editable':
+    case 'limited':
+      if (!isLogin) {
+        blockSourceView = true
+        disableControls()
+      } else {
+        blockSourceView = false
+        enableControls()
+      }
+      break
+    case 'locked':
+    case 'protected':
+    case 'private':
+      if (personalInfo.userid && window.owner && personalInfo.userid === window.owner) {
+        blockSourceView = false
+      } else {
+        blockSourceView = true
+        disableControls()
+      }
+      break
+  }
+}
+
+function disableControls () {
```

Comment:
Using more precise function name here.
Maybe can change `disableControls`to `disableModeChangeControls` and 
 `enableControls` to `enableModeChangeControls`